plugins/TouchIcon/class.touchicon.plugin.php

```diff
@@ -43,33 +60,45 @@ public function SettingsController_TouchIcon_Create($Sender) {
       $Sender->Permission('Garden.Settings.Manage');
       $Sender->AddSideMenu('dashboard/settings/touchicon');
       $Sender->Title(T('Touch Icon'));
-      
+
       if ($Sender->Form->AuthenticatedPostBack()) {
          $Upload = new Gdn_UploadImage();
          try {
             // Validate the upload
             $TmpImage = $Upload->ValidateUpload('TouchIcon', FALSE);
             if ($TmpImage) {
-               // Save the uploaded image
+               // Save the uploaded image.
+               $TouchIconPath ='banner/touchicon_'.substr(md5(microtime()), 16).'.png';
                $Upload->SaveImageAs(
                   $TmpImage,
-                  PATH_ROOT.'/uploads/TouchIcon/apple-touch-icon.png',
+                  $TouchIconPath,
                   114,
                   114,
                   array('OutputType' => 'png', 'ImageQuality' => '8')
                );
-               SaveToConfig('Plugins.TouchIcon.Uploaded', TRUE);
+               SaveToConfig('Garden.TouchIcon', $TouchIconPath);
             }
          } catch (Exception $ex) {
             $Sender->Form->AddError($ex->getMessage());
          }
-         
+
          $Sender->InformMessage(T('TouchIconSaved', "Your icon has been saved."));
       }
-      
-      $Sender->Render($this->GetView('touchicon.php'));      
+
+      $Sender->SetData('Path', $this->getIconPath());
+      $Sender->Render($this->GetView('touchicon.php'));
+   }
+
+   /**
+    * Get path to icon.
+    *
+    * @return string Path to icon
+    */
+   public function getIconPath() {
+      $Icon = C('Garden.TouchIcon') ? Gdn_Upload::Url(C('Garden.TouchIcon')) : self::DEFAULT_PATH;
```

Comment:
You have one case returning a url and the other returning a path. Won't that fail in subfolders?